### DISCUSSION TITLE
Thorlabs GVS212 galvo scanner with NI 6343

### DISCUSSION THREAD

**Discussion by kalosu** [Ulm-IQO/qudi-core#86] [Q&A] [answered]
Hello there Qudi community, 

I'm new to this world and I would like to ask for a bit of your help.

I am intending to build a fluorescence confocal imaging setup for measuring point emitters. 

I am intending to perform a scan over a sample substrate in X and Y with a Thorlabs GVS212 galvo scanner and I was wondering if it is possible to control this galvo scanner through a NI-USB 6343 DAQ. 

Do you have any experience with this? Would this be something supported by Qudi? 

Additionally, I would like to control a piezo z adjuster with a high NA objective mounted to it.  Would this also be supported by Qudi?

Sorry if my questions are not too "specific" but I am new to this world.  I'm trying to find some supporting architecture which would facilitate me to implement what I have intended.

In advance, thanks for your feedback and help!

**Comment by NicolasStaudenmaier** [comment, marked as answer]
Hi kalosu, 

I use the same NI-USB 6343 DAQ in my experiments with single NV centers. My scanning system is a 3D piezo stage. I don't see why it shouldn't work with a galvo scanner. If you control two axes of the galvo scanner with the NI card instead of a piezo stage it is equivalent. You just need some calibration in order to linearly transform the NI card's voltage output to the correct scanning range in meters. Piezo z-adjustment of the objective is identical and acquisition of the counts is done with the NI card as well (both digital and/or analog data). 

As a general remark, the scanning module in the [qudi-iqo-modules repository](https://github.com/Ulm-IQO/qudi-iqo-modules) supports scanning of arbitrary number of axes with individual voltage/scan ranges. If you have further questions feel free to ask.

**Comment by Neverhorst** [comment]
Hi @kalosu ,

I'm always happy to see new people using qudi. 

As @NicolasStaudenmaier already mentioned, it might be possible to use some or all of the existing hardware module implementations of the `qudi-iqo-modules` package for your use case.

However, the current hardware modules (and interfuse) used for confocal scanning with NI DACs was not designed with this use case in mind and it is possible you might encounter problems.
As a fallback solution you can always use the existing GUI and logic modules of the scanning probe toolchain and implement your own hardware module implementing the [`ScanningProbeInterface`](https://github.com/Ulm-IQO/qudi-iqo-modules/blob/e29d4680decc9359db91765358dc022703d9a6a0/src/qudi/interface/scanning_probe_interface.py#L29C20-L29C20).
In other words, you would need to create a new class inheriting this interface and implementing all abstract methods defined in the interface.
The rest of the scanning probe toolchain should then work with this new module. 

But for now, let's hope the existing module works for you right away.

**Comment by kalosu** [comment]
Hi @NicolasStaudenmaier , hi @Neverhorst ,

Thanks a lot for the nice feedback. At the end, I will be using some piezo actuators with a piezo controller from Thorlabs. I guess that this is also viable through Qudi. 

In any case, I will be giving it a try and if anything I will coming back to the forum!

Again thanks.
